commit 95f14b58f92a672669b41f1e4ce6c02f82155eca (refs/changes/20/4420/2)
Author: Alex Wilson <alex.wilson@joyent.com>
Date:   2018-06-22T15:06:36-07:00 (1 year, 4 months ago)
    
    MANTA-3785 waferlock allowing destroyed VMs
    MANTA-3786 waferlock doesn't deal with old VMs with no "ips" property on NICs

diff --git a/lib/sapi.js b/lib/sapi.js
index c9e4ef2..25915be 100644
--- a/lib/sapi.js
+++ b/lib/sapi.js
@@ -249,10 +249,15 @@ SapiPoller.prototype.state_runq_do = function (S) {
 		mod_assert.arrayOfObject(objs);
 
 		objs.forEach(function (vm) {
-			var ips = [];
+			if (vm.state === 'destroyed' || vm.destroyed)
+				return;
 
+			var ips = [];
 			vm.nics.forEach(function (nic) {
-				nic.ips.forEach(function (ip) {
+				var ipa = nic.ips;
+				if (ipa === undefined)
+					ipa = [nic.ip];
+				ipa.forEach(function (ip) {
 					var parts = ip.split('/');
 					ips.push(parts[0]);
 				});
